### PR TITLE
feat: Updates to schema and logic for removal of sources

### PR DESCRIPTION
**Pull request by lfvbarcus**
This MR removes sources key from schema and updates the code and testdata to accomodate the changes made. 
A new release will be created after merge.

### CODE REVIEW

File: src/reqstool/model_generators/requirements_model_generator.py [hunk 1]
```diff
@@ -309,15 +308,10 @@ def __parse_requirements(self, data):  # NOSONAR
         return r_reqs
 
     def __parse_requirements_reference(
```
Comment (lfvjimisola) [inline src/reqstool/model_generators/requirements_model_generator.py:310]: Do we need this method? Just to a list comprehension at L288 instead perhaps?

File: src/reqstool/resources/schemas/v1/requirements.schema.json [hunk 2]
```diff
@@ -250,11 +250,22 @@
...
-                    "type": "array",
-                    "items": {
-                        "$ref": "#/$defs/reference"
+                    "type": "object",
+                    "additionalProperties": false,
+                    "properties": {
+                        "requirement_ids": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "common.schema.json#/$defs/reqid",
```
Comment (lfvjimisola) [inline src/reqstool/resources/schemas/v1/requirements.schema.json:259]: is this a reqid or an urn_id or perhaps both? the def in common might need an update if it is in fact urn_id.

Comment (lfvbarcus) [reply src/reqstool/resources/schemas/v1/requirements.schema.json:259]: I have just reused the definition that we used previously. 
The definition for reqid in the common schema is just any string, and there are currently no definition set for urnid.

 This field is always interpreted a string, and if there is a ":" in it, reqstool will split it up as a urn_id .

I don't know it that is robust enough. What would you propose as an alternate solution?

Comment (lfvjimisola) [reply src/reqstool/resources/schemas/v1/requirements.schema.json:259]: Not sure. Let's write a ticket to follow-up on it. Or?

**Review by lfvjimisola — APPROVED** [review]

File: src/reqstool/resources/schemas/v1/requirements.schema.json (hunk 2 above)
Comment (lfvbarcus) [reply src/reqstool/resources/schemas/v1/requirements.schema.json:259]: yes, I'll do that!

**Pull request merged**